File: Brazil.pas
Program Brazil;
Var coluna, linha, i, j: integer;
Begin
    for coluna := 0 to 100 do begin
        for linha := 0 to 30 do begin
            gotoxy(coluna,linha);
            textcolor(green);
            writeln(#219);
        end;
    end;
    for i := 0 to 10 do begin
        linha := 5;
        coluna := 50 - i*3;
        linha := linha + i;
        if i > 0 then begin
            repeat
                gotoxy(coluna,linha);
                textcolor(yellow);
                writeln(#219);
                coluna := coluna + 1;
            until coluna = 50 + i*3;
        end else begin
            gotoxy(coluna,linha);
            textcolor(yellow);
            writeln(#219);
        end;
    end;
    for i := 0 to 10 do begin
        linha := 15;
        coluna := 20 + i*3;
        linha := linha + i;
        if i < 10 then begin
            repeat
                gotoxy(coluna,linha);
                textcolor(yellow);
                writeln(#219);
                coluna := coluna + 1;
            until coluna = 80 - i*3;
        end else begin
            gotoxy(coluna,linha);
            textcolor(yellow);
            writeln(#219);
        end;
    end;
    coluna := 49;
    linha := 15;
    i := 5;
    j := 1;
    repeat
        linha := 15 - i;
        coluna := 49 - j*3;
        repeat
            gotoxy(coluna,linha);
            textcolor(blue);
            writeln(#219);
            coluna := coluna + 1;
        until coluna = 49 + 3*(j+1);
        i := i - 1;
        j := j + 1;
    until i = 1;
    coluna := 37;
    linha := 13;
    repeat
        coluna := 37;
        repeat
            gotoxy(coluna,linha);
            textcolor(blue);
            writeln(#219);
            coluna := coluna+1;
        until coluna = 64;
        linha := linha + 1;
    until linha = 18;
    coluna := 40;
    linha := 18;
    i := 0;
    j := 1;
    repeat
        coluna := 40 + 3*i;
        repeat
            gotoxy(coluna,linha);
            textcolor(blue);
            writeln(#219);
            coluna := coluna + 1;
        until coluna = 64 - j*3;
        linha := linha + 1;
        i := i + 1;
        j := j + 1;
    until j = 4;
    for coluna := 37 to 63 do begin
        for linha := 14 to 16 do begin
            gotoxy(coluna,linha);
            textcolor(7);
            writeln(#219);
        end;
    end;
    gotoxy(42, 15);
    textbackground(white);
    textcolor(green);
    write('ORDEM E PROGRESSO');
End.

File: brazil.pl
line(0, ".", 0, 41).
line(1, ".", 0, 20).
line(1, "#", 20, 1).
line(1, ".", 21, 20).
line(2, ".", 0, 17).
line(2, "#", 17, 7).
line(2, ".", 24, 17).
line(3, ".", 0, 14).
line(3, "#", 14, 13).
line(3, ".", 27, 14).
line(4, ".", 0, 12).
line(4, "#", 12, 4).
line(4, "`", 16, 9).
line(4, "#", 25, 4).
line(4, ".", 29, 12).
line(5, ".", 0, 10).
line(5, "#", 10, 4).
line(5, "`", 14, 13).
line(5, "#", 27, 4).
line(5, ".", 31, 10).
line(6, ".", 0, 8).
line(6, "#", 8, 3).
line(6, "`", 11, 19).
line(6, "#", 30, 3).
line(6, ".", 33, 8).
line(7, ".", 0, 6).
line(7, "#", 6, 5).
line(7, " ORDEM E PROGRESSO ", 11, 1).
line(7, "#", 30, 5).
line(7, ".", 35, 6).
line(8, ".", 0, 8).
line(8, "#", 8, 3).
line(8, "`", 11, 19).
line(8, "#", 30, 3).
line(8, ".", 33, 8).
line(9, ".", 0, 10).
line(9, "#", 10, 4).
line(9, "`", 14, 13).
line(9, "#", 27, 4).
line(9, ".", 31, 10).
line(10, ".", 0, 12).
line(10, "#", 12, 4).
line(10, "`", 16, 9).
line(10, "#", 25, 4).
line(10, ".", 29, 12).
line(11, ".", 0, 14).
line(11, "#", 14, 13).
line(11, ".", 27, 14).
line(12, ".", 0, 17).
line(12, "#", 17, 7).
line(12, ".", 24, 17).
line(13, ".", 0, 20).
line(13, "#", 20, 1).
line(13, ".", 21, 20).
line(14, ".", 0, 41).
 
printrep(_, 0) :- !.
printrep(X, N) :- write(X), N1 is N-1, printrep(X, N1).

printflag() :- printline(0, 0).

printline(L, P) :-
    line(L, C, P, O),
    printrep(C, O),
    atom_length(C, CL),
    NP is P + O - 1 + CL,
    NP =< 41,
    line(L, _, NP, _),
    printline(L, NP), !.
printline(L, _) :- 
    nl, NL is L + 1,
    line(NL, _, _, _), !,
    printline(NL, 0).
printline(_, _) :- !.
